src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -331,6 +338,196 @@ def db_update(
+def _get_storage_key(
+        storage_account,
+        resource_group_name,
+        use_secondary_key):
+
+    # Get storage keys
+    client = get_mgmt_service_client(StorageManagementClient)
+    keys = client.storage_accounts.list_keys(
+        resource_group_name=resource_group_name,
+        account_name=storage_account)
+
+    # Choose storage key
+    index = 1 if use_secondary_key else 0
+    return keys.keys[index].value  # pylint: disable=no-member
+
+
+# Common code for updating audit and threat detection policy
+def _db_security_policy_update(  # pylint: disable=too-many-arguments
+        instance,
+        enabled,
+        storage_account,
+        storage_endpoint,
+        storage_account_access_key,
+        use_secondary_key):
+
+    # Validate storage endpoint arguments
+    if storage_endpoint is not None and storage_account is not None:
+        raise CLIError('--storage-endpoint and --storage-account cannot both be specified.')
+
+    # Set storage endpoint
+    if storage_endpoint is not None:
+        instance.storage_endpoint = storage_endpoint
+    if storage_account is not None:
+        storage_resource_group = _find_storage_account(storage_account)
+        instance.storage_endpoint = _get_storage_endpoint(storage_account, storage_resource_group)
+
+    # Set storage access key
+    if storage_account_access_key is not None:
+        # Access key is specified
+        instance.storage_account_access_key = storage_account_access_key
+    elif enabled:
+        # Access key is not specified, but state is Enabled.
+        # If state is Enabled, then access key property is required in PUT. However access key is
+        # readonly (GET returns empty string for access key), so we need to determine the value
+        # and then PUT it back. (We don't want the user to be force to specify this, because that
+        # would be very annoying when updating non-storage-related properties).
+        # This doesn't work if the user used generic update args, i.e. `--set state=Enabled`
+        # instead of `--state Enabled`, since the generic update args are applied after this custom
+        # function, but at least we tried.
+        if storage_account is None:
+            storage_account = _get_storage_account_name(instance.storage_endpoint)
+            storage_resource_group = _find_storage_account(storage_account)
+
+        instance.storage_account_access_key = _get_storage_key(
+            storage_account,
+            storage_resource_group,
+            use_secondary_key)
+
+
+# Update audit policy. Custom update function to apply parameters to instance.
+def db_audit_policy_update(  # pylint: disable=too-many-arguments
+        instance,
+        state=None,
+        storage_account=None,
+        storage_endpoint=None,
+        storage_account_access_key=None,
+        audit_actions_and_groups=None,
+        retention_days=None):
+
+    # Apply state
+    if state is not None:
```

Comment:
Yes, it will be validated due to `enum_choice_list` in params.py

(env) D:\git\azure-cli [datasecurity ≡]> az sql db audit-policy update --state adf
az sql db audit-policy update: error: argument --state: invalid choice: 'adf' (choose from 'Enabled', 'Disabled')